### ISSUE TITLE
[authorization] userToPrinciple using incorrect id/securityId

### ISSUE THREAD

**Issue by JohnPhoto** [strongloop/loopback-next#3707]
## Steps to reproduce
The `userToPrinciple` method in authZ breaks current `UserProfile` at the following line:
https://github.com/strongloop/loopback-next/blob/81890fce1e5cdaf1821557537164f23ba687386b/packages/authorization/src/authorize-interceptor.ts#L160

## Current Behavior
As you can see the method basically throws the `securityId` away (by setting it to the name) and tries to set the `id` (which no longer is valid in UserProfile to `securityId` on the principal.
So  from here on, `currentUserProfile` always has an `undefined` securityId.

## Expected Behavior

The principal which is used in authorizors `(  currentUser = authorizationCtx.principals[0];)` should have a working `securityId`.

tagging @raymondfeng @jannyHou as the usual suspects :) 

## Using:
```
├─┬ @loopback/authentication@3.0.0
│ └── @loopback/security@0.1.1  deduped
├─┬ @loopback/authorization@0.2.2
│ └── @loopback/security@0.1.1  deduped
└── @loopback/security@0.1.1
```

**Comment by dhmlau** [comment]
Thanks @JohnPhoto for tagging them. :) 

@jannyHou @raymondfeng , could you please take a look? Thanks.

**Comment by derdeka** [comment]
Related to #3766

**Comment by derdeka** [comment]
Fixed with #3807 ?

**Comment by derdeka** [comment]
@jannyHou @raymondfeng @JohnPhoto 
Is the issue resolved as #3807 is merged?

**Comment by dhmlau** [comment]
Closing as resolved.
